src/execution_plan/ops/op_delete.c

```diff
@@ -104,10 +107,25 @@ OpResult OpDeleteReset(OpBase *ctx) {
 void OpDeleteFree(OpBase *ctx) {
 	OpDelete *op = (OpDelete *)ctx;
 
-	_DeleteEntities(op);
+	if(op->deleted_nodes || op->deleted_edges) _DeleteEntities(op);
```

Comment:
I guess delete operations don't need to make the PropagateFree call because they perform all work within the free routine? It makes me a little nervous how the logical flow of the various operations are starting to deviate from each other.

Reply:
Correct and correct, I wish we had a single place where we dealt with all graph modifications:
1. Addition
2. Deletions
3. Update
and still be performant about it

Reply:
The bulk delete improvements you introduced recently were a major performance boost; I'm personally in favor of more complex interactions with GraphBLAS like that. Maybe hide our more complex logic in a `graph_internal.c` or something like that?